doc/api/esm.md

```diff
@@ -431,6 +431,46 @@ thrown:
 }
 ```
 
+#### Self-referencing a package using its name
+
+Self-referencing a package's name in an `import` is allowed. For example,
+let's assume the `package.json` is:
+
+```json
+// package.json
+{
+  "name": "a-package",
+  "exports": {
+    ".": "./main.mjs",
+    "./cjs": "./main.js"
```

Comment:
```js
  "exports": {
    ".": "./main.js",
    "./foo": "./foo.js"
```

I think my suggestion got swallowed up. Per conversation with @jkrems, I was suggesting we leave dual-mode out of this so as not to complicate the example. That means we need to use all-CommonJS sources so that the same `package.json` can work for a `require` example.

Reply:
It's not really dual mode, as it's not using conditional exports, but I see your point. I will change it to from "cjs" to "foo".